[12.0][MIG] hr_skill

alexey-pelykh opened the pull request.

Commit:
[ADD] adds hr-skill module

Commit:
[IMP] fix the copyright and update to reflect the new fields and views

Commit:
[IMP] don't need to instantiate a class in 7.0

Commit:
[IMP] add pot file

Commit:
[IMP] the domain for the employee skill ids is defined on the model instead on the form

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

dreispt committed:
Initialize 8.0 branch

dreispt committed:
Port Skills to v8

dreispt committed:
Enable child_of domain operator

yvaucher committed:
Add missing default oca icons

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[IMP] delete field for  view\skill type

Commit:
[IMP] display tag's name with parents, i.e. 1/2/3

Commit:
fix pep8

Commit:
[IMP] add skill_ids to skill form

[IMP] use "Name" instead "Display Name" in skill tree view

Commit:
[IMP] order skill by parent_left

Commit:
[IMP] allow select only skills without childs

Commit:
[FIX] access rights

Commit:
[IMP] show employees in skill tree

Commit:
bump 8.0.1.1.0

Commit:
add myself to contibutors

pedrobaeza committed:
[MIG] Make modules uninstallable

pedrobaeza committed:
[MIG] Rename manifest files

jlaloux committed:
[MIG] 10.0 porting hr_skill

[UPD] Move Skills menu into HR configuration and add widget m2m_tags in employee form view and add to contributors list

[UPD] Update copyright

[IMP] Add unit test to test nam_get function on hr_skill

Refactor code for pep-8

Add group to Skill menu

leemannd committed:
Fix imports & lost things in pull/300

Commit:
[MIG] hr_skill: Migration to 11.0

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update hr_skill.pot

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 95.0% (19 of 20 strings)

Translation: hr-11.0/hr-11.0-hr_skill
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_skill/pt_BR/

lfreeke commented:
When searching( via filters) on a name of a skill I get the following error: 

Employees -> Filters -> Add Custom filter -> Select Skills -> Contains -> [name of skill] -> Apply

Odoo Server Error
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/http.py", line 653, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/.repo_requirements/odoo/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/.repo_requirements/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/.repo_requirements/odoo/odoo/http.py", line 695, in dispatch
    result = self._call_function(**self.params)
  File "/.repo_requirements/odoo/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/.repo_requirements/odoo/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 938, in __call__
    return self.method(*args, **kw)
  File "/.repo_requirements/odoo/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 904, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 926, in do_search_read
    offset=offset or 0, limit=limit or False, order=sort or False)
  File "/.repo_requirements/odoo/odoo/models.py", line 4507, in search_read
    records = self.search(domain or [], offset=offset, limit=limit, order=order)
  File "/.repo_requirements/odoo/odoo/models.py", line 1553, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/.repo_requirements/odoo/odoo/models.py", line 4043, in _search
    query = self._where_calc(args)
  File "/.repo_requirements/odoo/odoo/models.py", line 3834, in _where_calc
    e = expression.expression(domain, self)
  File "/.repo_requirements/odoo/odoo/osv/expression.py", line 670, in __init__
    self.parse()
  File "/.repo_requirements/odoo/odoo/osv/expression.py", line 1021, in parse
    ids2 = [x[0] for x in comodel.name_search(right, domain or [], op2, limit=None)]
  File "/.repo_requirements/odoo/odoo/models.py", line 1635, in name_search
    return self._name_search(name, args, operator, limit=limit)
  File "/.repo_requirements/odoo/odoo/models.py", line 1648, in _name_search
    ids = self._search(args, limit=limit, access_rights_uid=access_rights_uid)
  File "/.repo_requirements/odoo/odoo/models.py", line 4039, in _search
    if expression.is_false(self, args):
  File "/.repo_requirements/odoo/odoo/osv/expression.py", line 225, in is_false
    elif token[1] == 'in' and not token[2]:
IndexError: string index out of range

alexey-pelykh commented:
Affected by #503

Commit:
[IMP] hr_skill 11.0.2.0.0

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update hr_skill.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

max3903 committed:
[IMP] Add uniqueness constraint

max3903 committed:
[IMP] Skill view

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (28 of 28 strings)

Translation: hr-11.0/hr-11.0-hr_skill
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_skill/de/

oca-travis committed:
[UPD] Update hr_skill.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (29 of 29 strings)

Translation: hr-11.0/hr-11.0-hr_skill
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_skill/de/

osi-scampbell committed:
[IMP] Active Smart Button

osi-scampbell committed:
[IMP] description

oca-travis committed:
[UPD] Update hr_skill.pot

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: hr-11.0/hr-11.0-hr_skill
Translate-URL: https://translation.odoo-community.org/projects/hr-11-0/hr-11-0-hr_skill/

max3903 committed:
[IMP] Add color to hr_skill

tbaden reviewed (commented):
Bug reported by @lfreeke is still there

alexey-pelykh commented:
@tbaden @lfreeke working on it

alexey-pelykh committed:
[MIG] hr_skill: Migration to 12.0

alexey-pelykh commented:
@tbaden @lfreeke redid the migration since v11 had more features

alexey-pelykh commented:
Bug fixed, all good for review.

lfreeke commented on hr_skill/views/hr_employee_skill.xml:
```diff
@@ -0,0 +1,74 @@
+            <form string="Employee Skill">
+                <group>
+                    <group>
+                        <field name="employee_id"/>
```
I don't think this field is needed. I think it is confusing for the user. 
If  add a skill on an employee form I would think it would be for that employee and I now randomly can choose another employee. 

Or is there another reason for this field?

alexey-pelykh replied:
This field is shown for another reason on that form, form itself should not be shown. Fixed in other way 👍

lfreeke reviewed (approved):
:+1: (functional test)

alexey-pelykh commented:
@feketemihai @max3903 please review if you don't mind

max3903 reviewed (approved):
LGTM. No test.

rafaelbn commented:
Please, if possible look at https://github.com/OCA/hr/issues/626. Thanks
